### PR TITLE
Create unified nav elements for Perf APIs.

### PR DESCRIPTION
**Pull request by jpmedley**

### CODE REVIEW

**Comment by jpmedley** [comment]
@chrisdavidmills

**Comment by chrisdavidmills** [comment]
Hi Joe,

I'm a bit confused here. Why have you added an entry to GroupData for this API, but also made a separate PerformanceSidebar Macro? The added GroupData is read and turned into a sidebar by the APIRef macro:

https://github.com/mozilla/kumascript/blob/master/macros/APIRef.ejs 

See https://developer.mozilla.org/en-US/docs/Web/API/WebVR_API for an example of how this is used.

I'll review your GroupData entry too.

File: macros/GroupData.json [hunk 1]
```diff
@@ -1258,6 +1258,68 @@
...
+            "overview":   [ "Performance API" ],
+            "interfaces": [ "Performance",
+                            "PerformanceEntry",
+                            "PerformanceFrameTiming",
+                            "PerformanceMark",
+                            "PerformanceMeasure",
+                            "PerformanceNavigationTiming",
+                            "PerformanceObserver",
+                            "PerformanceResourceTiming",
+                            "window.performance" ],
```
Comment (chrisdavidmills) [inline macros/GroupData.json]: Window.performance

File: macros/GroupData.json [hunk 2]
```diff
@@ -1258,6 +1258,68 @@
...
+                            "PerformanceNavigationTiming.domComplete",
+                            "PerformanceNavigationTiming.domContentLoadedEventEnd",
+                            "PerformanceNavigationTiming.domContentLoadedStart",
+                            "PerformanceNavigationTiming.domInteractive",
+                            "PerformanceNavigationTiming.loadEventEnd",
+                            "PerformanceNavigationTiming.loadEventStart",
+                            "PerformanceNavigationTiming.redirectCount",
+                            "PerformanceNavigationTiming.type",
+                            "PerformanceNavigationTiming.unloadEventEnd",
+                            "PerformanceNavigationTiming.unloadEVentStart",
```
Comment (chrisdavidmills) [inline macros/GroupData.json]: unloadEventStart

**Comment by jpmedley** [comment]
PTAL

**Comment by jpmedley** [comment]
@chrisdavidmills Is there anything else I need to do to this?

**Comment by chrisdavidmills** [comment]
Sorry Joe; I've still had a chance to go through this. I'll get to it soon, promise.

**Comment by jpmedley** [comment]
@chrisdavidmills

**Comment by chrisdavidmills** [comment]
Do you know what? I'm drowning in work, and having a terrible time finding the time to look at this. Sorry Joe.

@Elchi3 , this looks OK to me from a cursory glance. Can you give it a second look if you get the chance?

**Comment by jpmedley** [comment]
Thanks, guys!

Joe Medley | Technical Writer, Chrome DevRel | jmedley@google.com |
 816-678-7195
*If an API's not documented it doesn't exist.*

**Comment by Elchi3** [comment]
This question still stands. What's the motivation for {{PerformanceSidebar}} ?
> Why have you added an entry to GroupData for this API, but also made a separate PerformanceSidebar Macro?

**Comment by jpmedley** [comment]
Sorry for the turmoil. What I had in mind is slowly coming back to me. I
had intended to add more to what the macro showed after getting it sanity
checked.To be honest, I'm now thinking my reasoning was a bit shaky. Here's
the problem I thought I was solving.

The APIs listed in group data are all related yet many of them have
separate entries in groupdata, and separate overview pages. At the time I
opened this PR, I believed I was saving us the trouble of editing a few
hundred APIRef() macro calls (shaky reasoning).

Unless someone has another idea, let me take another shot at this using
just groupdata. Again, sorry for the turmoil.

Joe

Joe Medley | Technical Writer, Chrome DevRel | jmedley@google.com |
 816-678-7195
*If an API's not documented it doesn't exist.*

**Comment by Elchi3** [comment]
No worries, Joe. I think this should be good with just the GroupData entries.

(Refactoring the sidebars to be somewhat more coherent and solid is on my radar, but won't happen anytime soon. Also, we need to first find out how much they are used to judge if it is worth our time to improve this code).

**Comment by jpmedley** [comment]
I actually thought unifying the sidebar and creating a master overview page
would be a lot of bang for very few bucks. I personally think the ability
to test performance in the wild has enormous value.

Joe Medley | Technical Writer, Chrome DevRel | jmedley@google.com |
 816-678-7195
*If an API's not documented it doesn't exist.*

**Comment by jpmedley** [comment]
Performance navigation, take II. This is part 1 of 3.

**1. Create "Performance API" in GroupData.json.**
2. Add {{APIRef("Performance API")}} to relevant pages.
3. Remove other API groups from GroupData.json ("Performance Timeline API" for example.)

File: macros/GroupData.json [hunk 3]
```diff
@@ -1258,6 +1258,93 @@
...
         },
+        "Performance API": {
+            "overview":   [ "Performance API" ],
+            "guides":     [ { "url": "/en-US/docs/Web/API/Frame_timing_API",
+                              "title": "Frame Timing API" },
+                            { "url": "/en-US/docs/Web/API/Long_Tasks_API",
+                              "title": "Long Tasks API" },
+                            { "url": "/en-US/docs/Web/API/Navigation_timing_API",
+                              "title": "Navigation Timing API" },
+                            { "url": "/en-US/docs/Web/API/Navigation_timing_API/Using_Navigation_Timing",
```
Comment (Elchi3) [inline macros/GroupData.json]: https://developer.mozilla.org//en-US/docs/Web/API/Navigation_timing_API/Using_Navigation_Timing 
This page page looks useless. Are you planning to write something here? If not, I think it shouldn't be here for now.

File: macros/GroupData.json [hunk 4]
```diff
@@ -1258,6 +1258,93 @@
...
+                            "PerformanceResourceTiming.responseEnd",
+                            "PerformanceResourceTiming.responseStart",
+                            "PerformanceResourceTiming.secureConnectStart",
+                            "PerformanceResourceTiming.transferSize",
+                            "PerformanceResourceTiming.workerStart",
+                            "TaskAttributionTiming.containerId",
+                            "TaskAttributionTiming.containerName",
+                            "TaskAttributionTiming.containerSrc",
+                            "TaskAttributionTiming.containerType" ],
+            "events":     [ "Performance.onresourcetimingbufferfull" ]
```
Comment (Elchi3) [inline macros/GroupData.json]: This needs to be the name of the event "resourcetimingbufferfull" to create a link to the event page:
https://developer.mozilla.org/en-US/docs/Web/Events/resourcetimingbufferfull

**Comment by jpmedley** [comment]
PTAL

**Comment by Elchi3** [comment]
Almost! I meant `"events": [ "resourcetimingbufferfull" ]`

This is used to build https://developer.mozilla.org/en-US/docs/Web/Events/  `+ eventName` and the URL here is https://developer.mozilla.org/en-US/docs/Web/Events/resourcetimingbufferfull.

**Comment by jpmedley** [comment]
PTAL

**Pull request merged**